pkg/pillar/cmd/nim/nim.go

```diff
@@ -274,6 +277,38 @@ func (n *nim) run(ctx context.Context) (err error) {
 	// has lost /persist/status/nim/DevicePortConfigList
 	dpcAvailTimer := time.After(dpcAvailableTimeLimit)
 
+	if !n.enabledLastResort {
+		// Even if lastresort DPC is disabled by config, it can be forcefully used
+		// until NIM obtains any proper network config (from controller or bootstrap config
+		// or override/usb json).
+		//  * Before onboarding, we can easily determine if there is going to be any network
+		//    config available. If there is neither bootstrap device config nor any network
+		//    config override inside the /config partition or inside an (already plugged in)
+		//    USB stick, then device has no source of network configuration (currently)
+		//    available. The only exception is if the user is planning to insert USB stick
+		//    with usb.json *later*, but this cannot be predicted.
+		//    In order to not prolong onboarding in situations where the use of lastresort
+		//    is expected (ethernet + DHCP scenarios; often relied upon in lab/testing cases),
+		//    we will forcefully enable lastresort immediately if the above conditions for
+		//    missing network config source are satisfied. If the user later inserts a USB
+		//    stick with usb.json or the device onboards using lastresort and obtains a proper
+		//    config from the controller, the lastresort DPC will be unpublished (unless it is
+		//    enabled explicitly by config - by default it is disabled).
+		//  * After onboarding, it is expected that device always keeps and persists
+		//    at least the latest and the last working DPC, so it should never run out of
+		//    network configurations. But should there be a persisted data corruption caused
+		//    by an ungraceful reboot/crash and all DPCs disappear, NIM will forcefully
+		//    enable lastresort if it still has no network configuration available even one
+		//    minute past boot (see dpcAvailTimer). But this should be now pretty much unreachable
+		//    after we implemented backup files for persisted pubsub messages.
+		if !n.isDeviceOnboarded() &&
+			len(n.listPublishedDPCs(runDevicePortConfigDir)) == 0 &&
+			!fileutils.FileExists(n.Log, types.BootstrapConfFileName) {
+			n.forceLastResort = true
```

Comment:
Should we set waitForLastResort in this case as well as below?
Or does n.enableLastResort get set as part of the toggle function?

Reply:
`n.enabledLastResort` is set inside the function - added comment to make it clear.